### DISCUSSION TITLE
How can i use Object.groupBy() ?

### DISCUSSION THREAD

**Discussion by satu-jovanhartono** [vercel/next.js#68317] [Help] [answered]
### Summary

typescript added Object.groupBy in 5.4 release
https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/#objectgroupby-and-mapgroupby

i quickly upgraded my nextjs project to typescript 5.4, but getting this error message on my terminal instead.

<img width="371" alt="Screenshot 2024-07-30 at 15 29 27" src="https://github.com/user-attachments/assets/c4e0d5f9-5229-442b-8e5f-629f4c306fce">

any clue?

**Comment by icyJoseph** [comment, marked as answer]
Assuming, this in Node environment, GroupBy is only available in Node 21+

**Reply by satu-jovanhartono** [reply]
wow, thanks for the quick reply. i just realized that i am loading a component that using `Object.groupBy` inside `useMemo()`. the thing is, `useMemo` is also rendered on the server, i am able to remove the error message from the terminal by dynamically import the component with `ssr: false`

by the way, is there any docs where can i see the list of ES features supported by node version?

**Reply by icyJoseph** [reply]
You can do a by feature look up at `MDN`, for example, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy, or by system look up here, https://compat-table.github.io/compat-table/es2016plus/
